[MPI] ImplicitTriangulation: Fixes for Cell IDs in 2D

m-s-will opened the pull request:
As an extension of #933 , this PR moves the division to `triangleToPosition2d` itself, to fix some lingering issues with global / local cell id computation.

m-s-will committed:
[MPI] Moved division for 2d to function itself

julien-tierny commented:
looks good to me.

@pierre-guillou is this OK with you? (cf https://github.com/topology-tool-kit/ttk/pull/933)

julien-tierny commented:
@pierre-guillou some examples are failing under windows. I suspect this might be a consequence of https://github.com/topology-tool-kit/ttk/pull/934 or https://github.com/topology-tool-kit/ttk/pull/935

what do you think?
(I can simply roll these back to give it a try)

pierre-guillou commented:
I don't think there is an issue with the CI.
If we look at what failed, it was either the `imageProcessing` and/or `timeTracking` Python scripts, both have 2D regular grids as input.

julien-tierny commented:
good catch. the CI is running with MPI support with only one process if I'm not mistaken.
some pre-conditioning is enabled while it should not?

m-s-will commented:
I was checking for this and saw that sometimes preconditioning stuff is encapsulated with `isRunningWithMPI`, some with `hasInitializedMPI` and some not at all. Is this due to functional differences depending on the amount of ranks or was this an oversight?

![Bildschirm­foto 2023-05-10 um 14 06 46](https://github.com/topology-tool-kit/ttk/assets/57011129/6e3c2482-b353-4012-a9f7-efbab54ccb96)
![Bildschirm­foto 2023-05-10 um 14 07 16](https://github.com/topology-tool-kit/ttk/assets/57011129/3266c5e8-2a67-4732-bc34-f2f75c04f7cf)
![Bildschirm­foto 2023-05-10 um 14 09 14](https://github.com/topology-tool-kit/ttk/assets/57011129/6c0dd382-59bb-40b5-a5d9-427a8f80910f)
![Bildschirm­foto 2023-05-10 um 14 09 11](https://github.com/topology-tool-kit/ttk/assets/57011129/8e967717-13c8-448b-877c-edb496b04bb0)

pierre-guillou commented:
With these changes but without the MPI support, my laptop tries to overflow my RAM + swap when launching ParaView on the two offending state files (everything works without).

pierre-guillou commented:
I don't think we should modify `triangleToPosition2d`. I fear this breaks a lot of assumptions in the `ImplicitTriangulation` module (the method is used in `ImplicitPreconditions` then the result coordinates are accessed through `getTriangleCoords`).

m-s-will committed:
Moved division back out, added missing division

m-s-will commented:
Yes I reverted it back and added the missing divisions to the correct places, the state files should not fail anymore and the Cell id conversions are correct now too.

pierre-guillou commented:
You might want to remove the leftover comment in `ImplicitTriangulation.h` to reduce the time spent in recompiling all the dependent modules.

m-s-will committed:
Moved leftover comment

julien-tierny commented:
great stuff. thanks a lot to the both of you.
